packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/javascript/guide.md

```diff
@@ -1,3 +1,406 @@
+
+### Sign In to Okta with a Custom Dashboard
+
+If you are signing your users in to Okta, but you don't want to use the Okta dashboard, then you can change the redirect URL to point to your custom portal instead.
+
+```javascript
+function success(res) {
+  if (res.status === 'SUCCESS') {
+    res.session.setCookieAndRedirect('https://example.com/dashboard');
+  }
+}
+```
+
+### Sign In to Your Application
+
+If want to use the Widget to sign in to your own application instead of Okta, you need to [set-up a custom Authorization Server](/docs/guides/customize-authz-server/) in Okta.
+
+### Server-side Web Application using "authorization_code" flow
+
+```javascript
+
+const signIn = new OktaSignIn({
+  baseUrl: 'https://${yourOktaDomain}',
+  el: '#widget-container',
+  clientId: '${clientId}',
+  // must be in the list of redirect URIs enabled for the OIDC app
+  redirectUri: '${redirectUri}',
+  authParams: {
+    issuer: 'https://${yourOktaDomain}/oauth2/default',
+    pkce: false,
+    responseType: ['code']
+  }
+});
+
+// A query parameter named `code` will be passed to the login redirect URI
+// This should be handled by server-side code. The code can be exchanged for tokens
+signIn.showSignInAndRedirect();
+
+```
+
+### SPA or Native Application using PKCE
+
+```javascript
+
+const signIn = new OktaSignIn({
+  baseUrl: 'https://${yourOktaDomain}',
+  el: '#widget-container',
+  clientId: '${clientId}',
+  // must be in the list of redirect URIs enabled for the OIDC app
+  redirectUri: '${redirectUri}',
+  authParams: {
+    issuer: 'https://${yourOktaDomain}/oauth2/default'
+  }
+});
+
+// SPA and Native apps using PKCE can receive tokens directly without any redirect
+signIn.showSignInToGetTokens().then(function(tokens) {
+  // store/use tokens
+});
+
+```
+
+Here is an example of some front-end code that could use this token:
+
+```javascript
+function callMessagesApi() {
+  const accessToken = signIn.authClient.getAccessToken();
+
+  if (!accessToken) {
+    // This means that the user is not logged in
+    return;
+  }
+
+  // Make a request using jQuery
+  $.ajax({
+    // Your API or resource server:
+    url: 'http://localhost:8080/api/messages',
+    headers: {
+      Authorization: 'Bearer ' + accessToken.accessToken
+    },
+    success: function(response) {
+      // Received messages!
+      console.log('Messages', response);
+    },
+    error: function(response) {
+      console.error(response);
+    }
+  });
+}
+```
+
+### Handling Errors
+
+The Widget render function either results in a success or an error. The error function is called when the Widget has been initialized with invalid config options, or has entered a state it cannot recover from.
```

Comment:
```suggestion
The Widget render function either results in a success or an error. The error function is called when the Widget is initialized with invalid config options or entered a state it can't recover from.
```